src/closed.jl

```diff
@@ -0,0 +1,54 @@
+"""
+Defines the ClosedSet type.
+
+A ClosedSet(left, right) is an interval set where both bounds are closed i.e. the range is [left, right]
```

Comment:
"Defines the ClosedSet type" is probably unnecessary since it should (hopefully) be obvious to a user that the type constructor constructs the type. :smile: 

I'd recommend putting backticks around `ClosedSet(left, right)` and perhaps rephrasing a bit as: "A `ClosedSet(left, right)` is an interval set that includes both its upper and lower bounds. In mathematical notation, the constructed range is `[left, right]`."

I say "in mathematical notation" there so as to avoid confusion with the syntax for arrays.